server/schedule/placement/fit.go

```diff
@@ -52,6 +52,52 @@ func (f *RegionFit) IsCached() bool {
 	return f.mu.cached
 }
 
+// Replace return true if the replacement store is fit all constraints and isolation score is not less than the origin.
+func (f *RegionFit) Replace(srcStoreID uint64, dstStore *core.StoreInfo, region *core.RegionInfo) bool {
+	fit := f.getRuleFitByStoreID(srcStoreID)
+	// check the target store is fit all constraints.
+	if fit == nil || !MatchLabelConstraints(dstStore, fit.Rule.LabelConstraints) {
+		return false
+	}
+
+	// the members of the rule are same, it shouldn't check the score.
+	if fit.contain(dstStore.GetID()) {
+		return true
+	}
+
+	peers := make([]*fitPeer, 0, len(fit.Peers))
+	for _, p := range fit.Peers {
+		peer := &fitPeer{
+			Peer:     p,
+			isLeader: region.GetLeader().GetId() == p.GetId(),
+			store:    getStoreByID(f.regionStores, p.GetStoreId()),
+		}
+		if p.GetStoreId() == srcStoreID {
+			peer.store = dstStore
+		}
+		peers = append(peers, peer)
+	}
+	// Sort peers to keep the match result deterministic.
+	sort.Slice(peers, func(i, j int) bool {
+		// Put healthy peers in front of priority to fit healthy peers.
+		si, sj := stateScore(region, peers[i].GetId()), stateScore(region, peers[j].GetId())
+		return si > sj || (si == sj && peers[i].GetId() < peers[j].GetId())
+	})
```

Comment:
This part of the code is almost the same as the fitWorker, perhaps, we can abstract it into a function.